comp/src/main/java/frc/robot/subsystems/Intake/IntakeSubsystem.java

```diff
@@ -17,44 +22,100 @@
 
 public class IntakeSubsystem extends SubsystemBase {
     private SparkFlex motor = new SparkFlex(23, MotorType.kBrushless);
+    SparkFlexConfig motorConfig = new SparkFlexConfig();
+    // private boolean hasAlgae = false;
     private LaserCan intakeDist = new LaserCan(26);
+    private double speed = 0;
+    MedianFilter distanceFilter = new MedianFilter(10);
+    private double distMm;
+    private boolean ejecting = false;
 
-    public IntakeSubsystem() {
+    @Override
+    public void periodic() {
+        if (hasAlgae() && !ejecting) {
+            motor.set(speed / 8);
+        } else {
+            motor.set(speed);
+        }
+
+        distMm = distanceFilter.calculate(getRawSensorValue());
+        // if (motor.getOutputCurrent() > 60) {
+        // motor.set(speed / 8);
+        // hasAlgae = true;
+        // } else {
+        // motor.set(speed);
+        // hasAlgae = false;
+        // }
+
+        /*
+         * to do:
+         * 0 try distance
+         * 0 make distance work
+         * O fine tune algae descore w/ abs encoder
+         * O make autoalign and descore
+         * O try barge
+         * O try processor
+         */
+    }
+
+    private void setCurrentLimit(int limit, int free) {
+        motorConfig.smartCurrentLimit(limit, free);
+        motor.configureAsync(motorConfig, ResetMode.kNoResetSafeParameters, PersistMode.kNoPersistParameters);
+    }
 
+    public IntakeSubsystem() {
+        setCurrentLimit(80, 80);
     }
 
     public void intake(double speed) {
-        motor.set(speed);
+        this.speed = speed;
     }
 
     public void outtake(double speed) {
-        motor.set(speed);
+        this.speed = speed;
+    }
+
+    public void outtakeAlgae(double speed) {
+        this.speed = -speed;
+        ejecting = true;
     }
 
     public void backwards(double speed) {
-        motor.set(-speed);
+        this.speed = -speed;
     }
 
     public void stop() {
-        motor.set(0);
+        speed = 0;
+        ejecting = false;
     }
 
     public double distMm() {
+        return distMm;
+    }
+
+    public double getRawSensorValue() {
         var dist = intakeDist.getMeasurement();
-        if (dist == null) {
+        if (dist == null && intakeDist.getMeasurement().status == intakeDist.LASERCAN_STATUS_OUT_OF_BOUNDS) {
```

Comment:
This can never be true and in fact would throw an error when dist is null because if it is, then trying to `.status` from `intakeDist.getMeasurement()` would throw a `NullPointerException`.